src/macrobrowserui.cpp

```diff
@@ -51,20 +57,16 @@ MacroBrowserUI::~MacroBrowserUI()
         networkManager->deleteLater();
         networkManager=nullptr;
     }
-    foreach(QList<QTableWidgetItem *>lst,itemCache){
-        foreach(auto *item,lst){
-            delete item;
-        }
-    }
+    treeWidget->~QTreeWidget();
```

Comment:
this looks odd. Why call the destructor directly ?

Reply:
Thanks for reviewing. [From manual: Destroys the tree widget and all its items.](https://doc.qt.io/qt-6/qtreewidget.html#dtor.QTreeWidget)
So I assume, that this way the loop is not needed. When I assume that closing the dialog destroys all elements, we should not even need deleting the treeWidget, right? Then why has there been the loop? What should I do?

Reply:
the loop is not needed as it was only necessary for the cache.
to destruct an object, call "delete treewidget"